cep-20.2.md

```diff
@@ -0,0 +1,650 @@
+    # preferred_env_executable_paths: list
+
+    # marked as "still experimental"
+    # pin_depends: Enum<"record" | "strict">
+    # overlinking_ignore_patterns: [glob]
+
+    # defaults to patchelf (only cudatoolkit is using `lief` for some reason)
+    # rpaths_patcher: None
+```
+
+## Source section
+
+```yaml
+source: [SourceElement]
+```
+
+where the different source elements are defined as follows.
+
+### URL source
+
+```yaml
+# url pointing to the source tar.gz|zip|tar.bz2|...
+url: url
+# destination folder in work directory
+folder: path
+# hash of the file
+sha256: hex string
+# absolute or relative path from recipe file
+patches: [path]
+# removed
+# fn: string
+# md5: hex string
+# sha1: hex string
+```
+
+### Local source
+
+```yaml
+# file, absolute or relative to recipe file
+path: path
+# destination folder
+folder: path
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Git source
+
+```yaml
+# URL to the git repository or path to local git repository
+git_url: url | path
+# revision to checkout to (commit or tag)
+git_rev: string
+git_depth: signed integer (defaults to -1 -> not shallow)
+```
+
+### Removed source definitions
+
+SVN and HG (mercury) source definitions are removed as they are not relevant anymore.
+
+## Requirements section
+
+```yaml
+requirements:
+  # build time dependencies, in the build_platform architecture
+  build: [MatchSpec]
+  # dependencies to link against, in the target_platform architecture
+  host: [MatchSpec]
+  # the section below is copied into the index.json and required at package installation
+  run: [MatchSpec]
+  # constrain optional packages
+  run_constrained: [MatchSpec]
+```
+
+## Test section
+
+Note: the test section also has a weird implicit behavior with `run_test.sh`, `run_test.bat` as well as `run_test.py` and `run_test.pl` script files
+that are run as part of the test.
+
+This is the current state of the Test section.
+
+```yaml
+test:
+  # files (from recipe directory) to include with the tests
+  files: [glob]
+  # files (from the work directory) to include with the tests
+  source_files: [glob]
+  # requirements at test time, in the target_platform architecture
+  requires: [MatchSpec]
+  # commands to execute
+  commands: [string]
+  # imports to execute with python (e.g. `import <string>`)
+  imports: [string]
+  # downstream packages that should be tested against this package
+  downstreams: [MatchSpec]
+```
+
+### Proposal for new test section:
+
+```
+test: [TestElement]
+```
+
+where the different test elements are defined as follows:
+
+#### Command test element
+
+```yaml
+# script to execute
+script: string | [string]
+# optional extra requirements
+extra_requirements:
+  # extra requirements with build_platform architecture (emulators, ...)
+  build: [MatchSpec]
+  # extra run dependencies
+  run: [MatchSpec]
+# extra files to add to the package for the test
+# Alternatively, use ${{ RECIPE_DIR }} or ${{ SOURCE_DIR }} and a single list?
```

Comment:
I like this better, imo.

Reply:
I have mixed feelings. I always found `files` and `source_files` ambiguous. Your proposed format would help. I also like your alternatives, but I feel like it could cause ambiguity. For example, if we allow `${{ RECIPE_DIR }}`, does it mean that we can also do `/absolute/path/to/file`?